pkg/server/server.go

```diff
@@ -93,6 +93,17 @@ func WriteResponse(
 	c.JSON(statusCode, responseBody)
 }
 
+//WriteResponseWithRequestID writes a response and adds a request id header
+func WriteResponseWithRequestID(
+	c *gin.Context,
+	statusCode int,
+	requestID string,
+	responseBody interface{},
+) {
+	c.Header("X-Request-Id", requestID)
```

Comment:
x-request-id is usually populated by client, and not by server. does x-correlation-id fit better? 
https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#cite_note-stackoverflow1-39 
https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#cite_note-stackoverflow1-40
https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#cite_note-stackoverflow1-41 
https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#cite_note-stackoverflow1-42

Reply:
yes I agree